config/validate/validate.go

```diff
@@ -192,6 +211,46 @@ func validateStruct(vObj reflect.Value, ast astnode.AstNode, source []byte, chec
 
 		sub_report := Validate(f.Value, sub_node, src, checkUnusedKeys)
 		sub_report.AddPosition(line, col, highlight)
+
+		// duplicate checking time
+		if f.Value.Kind() == reflect.Slice {
+			// get the correct list of dups
+			dupListKey := f.Type.Name
+			if k, ok := mergedKeys[dupListKey]; ok {
+				dupListKey = k
+			}
+			if dupLists[dupListKey] == nil {
+				dupLists[dupListKey] = map[string]struct{}{}
+			}
+
+			for i := 0; i < f.Value.Len(); i++ {
+				key := ""
+				if f.Value.Index(i).Kind() == reflect.String {
```

Comment:
Is there ever a case where duplicate strings can/should be allowed? For instance, could a duplicate entry in say `FilesystemOptions` be valid but disallowed in this case (maybe multiple specifications of a parameter name)?

As well, is the `Append` rework going to handle cases for `sshAuthorizedKeys` where it won't introduce duplicates if the same SSH key is re-specified in appended configs (like we've seen in the past from the MCO case)? If not, then this would probably break them.

Reply:
`mdadm` commands come to mind. Will add something to allow disabling checking on certain lists.

The Append rework will, in the case of duplicate keys, result in a config with the key only once. Technically the child config's key will overwrite the parent's but it's the same key so it doesn't really matter.